euphoria-core/src/main/java/cz/seznam/euphoria/core/client/operator/ReduceByKey.java

```diff
@@ -391,7 +470,8 @@ public void add(E element) {
       if (v == null) {
         this.storage.set(element);
       } else {
-        this.storage.set(this.reducer.apply(Arrays.asList(v, element)));
+        this.reducer.apply(Arrays.asList(v, element), context);
+        this.storage.set(context.getValue());
```

Comment:
the `context` should probably be cleared at this point. it seems there is no reason to further hard reference the "value" after storing it in the backend storage. it could be GC'd right away.